synapse/cores/common.py

```diff
@@ -291,6 +302,144 @@ def isRuntForm(self, prop):
         '''
         return prop in self.runt_forms
 
+    def getCorePath(self, *paths):
+        '''
+        Construct a path relative to the cortex metadata dir (or None).
+
+        Args:
+            *paths ([str,]): A set of path elements
+
+        Returns:
+            (str):  The full path ( or None ).
+        '''
+        dirn = self.getConfOpt('dir')
+        if dirn is None:
+            return None
+
+        return s_common.genpath(dirn, *paths)
+
+    def reqCorePath(self, *paths):
+        '''
+        Use getCorePath and raise if dir is not set.
+
+        Raises:
+            NoSuchOpt
+        '''
+        retn = self.getCorePath(*paths)
+        if retn is None:
+            raise s_common.ReqConfOpt(name='dir', mesg='reqCorePath requires a cortex dir')
+        return retn
+
+    def getCoreFifo(self, name):
+        '''
+        Return a Fifo object by name.
+
+        Args:
+            name (str): The :name of the syn:fifo node.
+
+        Returns:
+            (Fifo): The Fifo object.
+        '''
+        return self._core_fifos.gen(name)
+
+    def putCoreFifo(self, name, item):
+        '''
+        Add an item to a cortex fifo.
+
+        Args:
+            name (str): The syn:fifo:name of the fifo.
+            item (obj): The object to put in the fifo.
+        '''
+        self.reqperm(('fifo:put', {'name': name}))
+        fifo = self._core_fifos.gen(name)
+        fifo.put(item)
+
+    def ackCoreFifo(self, name, seqn):
+        '''
+        Acknowledge transmission of fifo items.
+
+        Args:
+            name (str): The syn:fifo:name of the fifo.
+            nseq (int): The next expected sequence.
+        '''
+        self.reqperm(('fifo:ack', {'name': name}))
+        fifo = self._core_fifos.gen(name)
+        fifo.ack(seqn)
+
+    def _onFifoSub(self, mesg):
+        name = mesg[1].get('name')
+        xmit = self._getTeleFifoXmit(name)
+        self.subCoreFifo(name, xmit=xmit)
+
+    def _onFifoAck(self, mesg):
+        name = mesg[1].get('name')
```

Comment:
This calll code is untested which is weird since we have the ``Proxy._onFifoXmit`` called during tests which should fire a ``fifo:ack`` event